drivers/staging/vc04_services/interface/vchiq_arm/vchiq_core.c

```diff
@@ -2407,6 +2407,11 @@ vchiq_init_state(VCHIQ_STATE_T *state, VCHIQ_SLOT_ZERO_T *slot_zero,
 		return VCHIQ_ERROR;
 	}
 
```

Comment:
This check made more sense on the VPU which could be both master and slave. Since this is called from .probe the checking could be remove. By checking for the error you are suggesting it could happen, whereas my use of BUG was more a statement that it can't.